native/core/src/parquet/parquet_support.rs

```diff
@@ -382,14 +393,15 @@ pub(crate) fn prepare_object_store_with_configs(
         &url[url::Position::BeforeHost..url::Position::AfterPort],
     );
 
-    let (object_store, object_store_path): (Box<dyn ObjectStore>, Path) = if scheme == "hdfs" {
-        parse_hdfs_url(&url)
-    } else if scheme == "s3" {
-        objectstore::s3::create_store(&url, object_store_configs, Duration::from_secs(300))
-    } else {
-        parse_url(&url)
-    }
-    .map_err(|e| ExecutionError::GeneralError(e.to_string()))?;
+    let (object_store, object_store_path): (Box<dyn ObjectStore>, Path) =
+        if is_hdfs_scheme(&url, object_store_configs) {
```

Comment:
There is a little gotcha here when the scheme is s3a. In s3a's case, we replace s3a with s3 so that we can use the native object store implementation (https://github.com/apache/datafusion-comet/blob/60776f25038750848b26c7fadf78ca11ead71a3d/native/core/src/parquet/parquet_support.rs#L373). If the user has s3a in the list of hdfs urls because they want to use the hadoop-aws implementation, then they will still end up with the native implementation.

Reply:
Thanks, I made some adjustments, could you take another look?